sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3192,6 +3192,33 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession {
       checkAnswer(df3, Array(Row(new java.math.BigDecimal("0.100000000000000000000000100"))))
     }
   }
+
+  test("SPARK-29213 Make it consistent when get notnull output and generate null " +
```

Comment:
We can simply describe the bug in the test name: `SPARK-29213: FilterExec should not throw NPE`